src/main/java/net/minecraftforge/items/SlotItemHandler.java

```diff
@@ -126,7 +126,9 @@ public int getItemStackLimit(@Nonnull ItemStack stack)
     @Override
     public boolean canTakeStack(EntityPlayer playerIn)
     {
-        return !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        boolean vanillaResult = !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        boolean result = net.minecraftforge.event.ForgeEventFactory.onCanTakeItem(playerIn, this.getStack(), vanillaResult);
+        return result && vanillaResult;
```

Comment:
If `vanillaResult` here is `false`, is there a point in firing the event?
I think we should carefully consider if this is the right logic, and if it is we should make it something that avoids firing when it's not required, like
`return vanillaResult && net.minecraftforge.event.ForgeEventFactory.onCanTakeItem(playerIn, this.getStack(), vanillaResult);`

Reply:
I thought about that too. But then it occurred to me that a mod maker might want to override the `false` vanilla result and force the item to be picked up. In that case should we just remove `vanillaResult` from the return statement? If not then I think making the event only trigger when it is true also works.

Reply:
Any thoughts?

Reply:
Sounds good. Besides removing `vanillaResult` from that return statement, is there anything else that we need to do?